test/services/carrier_account.test.js

```diff
@@ -76,22 +94,50 @@ describe('CarrierAccount Service', function () {
   });
 
   it('updates a carrier account', async function () {
+    // Create a temporary carrier account
+    const carrierAccount = await this.client.CarrierAccount.create(Fixture.basicCarrierAccount());
+
     const testDescription = 'My custom description';
+    const updateParams = {
+      description: testDescription,
+    };
 
-    const carrierAccount = await this.client.CarrierAccount.create(Fixture.basicCarrierAccount());
+    // Update the carrier account
+    await this.client.CarrierAccount.update(carrierAccount.id, updateParams);
 
-    await this.client.CarrierAccount.retrieve(carrierAccount.id).then(async () => {
-      const params = {};
-      params.description = testDescription;
-      const updatedCarrierAccount = await this.client.CarrierAccount.update(
-        carrierAccount.id,
-        params,
-      );
-
-      expect(updatedCarrierAccount).to.be.an.instanceOf(CarrierAccount);
-      expect(updatedCarrierAccount.id).to.match(/^ca_/);
-      expect(updatedCarrierAccount.description).to.equal(testDescription);
-    });
+    // Retrieve the updated carrier account
+    const updatedCarrierAccount = await this.client.CarrierAccount.retrieve(carrierAccount.id);
+
+    expect(updatedCarrierAccount).to.be.an.instanceOf(CarrierAccount);
+    expect(updatedCarrierAccount.id).to.match(/^ca_/);
+    expect(updatedCarrierAccount.description).to.equal(testDescription);
+
+    // Remove the carrier account once we have tested it so we don't pollute the account with test accounts
+    await this.client.CarrierAccount.delete(carrierAccount.id);
+  });
+
+  it('updates a UPS carrier account', async function () {
+    // Create a temporary UPS carrier account
+    const params = {
+      type: 'UpsAccount',
+      account_number: '123456789',
+    };
+    const carrierAccount = await this.client.CarrierAccount.create(params);
+
+    const testAccountNumber = '987654321';
+    const updateParams = {
+      account_number: testAccountNumber,
+    };
+
+    // Update the carrier account
+    await this.client.CarrierAccount.update(carrierAccount.id, updateParams);
+
+    // Retrieve the updated carrier account
+    const updatedCarrierAccount = await this.client.CarrierAccount.retrieve(carrierAccount.id);
+
+    expect(updatedCarrierAccount).to.be.an.instanceOf(CarrierAccount);
+    expect(updatedCarrierAccount.id).to.match(/^ca_/);
+    // expect(updatedCarrierAccount.credentials["account_number"]).to.equal(testAccountNumber); // TODO: Re-enable assertion when API updates number properly
```

Comment:
```suggestion
    // expect(updatedCarrierAccount.fields.credentials["account_number"]).to.equal(testAccountNumber); // TODO: Re-enable assertion when API updates number properly
```

There is another layer of key to access credentials, and I don't think we need this comment

Reply:
`fields` is different than `credentials`. For now, both are empty because the API is not returning the proper data, hence the comment: 
<img width="500" alt="image" src="https://github.com/EasyPost/easypost-node/assets/17054780/520dc42d-e93c-48d6-821f-4e41d9925584">

Reply:
let's remove this until it's a for-sure thing they're going to be fixing this. I'm not sure we can rely on a timetable from data.